src/Plugin/Hal.php

```diff
@@ -548,6 +562,10 @@ public function renderEntity(Entity $halEntity, $renderEntity = true)
             $entity = array();
         }
 
+        if ($this->maxDepth and $depth > $this->maxDepth) {
```

Comment:
I actually have a concern here: `maxDepth` is never getting reset. This means that subsequent entities that are rendered have to adhere to the `maxDepth` set by a previous entity, even if they do not actually specify a max depth themselves. Since `maxDepth` is only used for comparisons, and never incremented/decremented, this could be problematic.

I think this could be resolved by adding the line:

```php
if (0 === $depth && $this->maxDepth) {
    $this->maxDepth = 0;
}
```

prior to returning from the method. However, we also definitely need a test to ensure the behavior I indicated does not occur.